src/vs/workbench/contrib/tasks/browser/runAutomaticTasks.ts

```diff
@@ -35,6 +36,11 @@ export class RunAutomaticTasks extends Disposable implements IWorkbenchContribut
 		@INotificationService private readonly _notificationService: INotificationService) {
 		super();
 		this._tryRunTasks();
+		this._register(this._workspaceTrustManagementService.onDidChangeTrust(async trusted => {
+			if (trusted) {
+				await this._tryRunTasks();
+			}
+		}));
```

Comment:
The trust state is inconsistent in the ctor vs the event, suggestion:

```suggestion
		if (this._workspaceTrustManagementService.isWorkspaceTrusted()) {
			this._tryRunTasks();
		}
		this._register(this._workspaceTrustManagementService.onDidChangeTrust(async trusted => {
			if (trusted) {
				await this._tryRunTasks();
			}
		}));
```